order/src/main/java/com/yas/order/service/CheckoutService.java

```diff
@@ -33,70 +35,94 @@
 
-        Checkout savedCheckout = checkoutRepository.save(checkout);
+        checkout = setCheckoutItems(checkout, checkoutPostVm);
 
-        CheckoutVm checkoutVm = checkoutMapper.toVm(savedCheckout);
-        if (CollectionUtils.isEmpty(checkoutPostVm.checkoutItemPostVms())) {
-            return checkoutVm;
-        }
+        checkout = checkoutRepository.save(checkout);
 
-        List<CheckoutItem> checkoutItemList = checkoutPostVm.checkoutItemPostVms()
-            .stream()
-            .map(checkoutItemPostVm -> {
-                CheckoutItem item = checkoutMapper.toModel(checkoutItemPostVm);
-                item.setCheckoutId(savedCheckout.getId());
-                return item;
-            })
-            .toList();
+        CheckoutVm checkoutVm = checkoutMapper.toVm(checkout);
+        Set<CheckoutItemVm> checkoutItemVms = checkout.getCheckoutItems()
+                .stream()
+                .map(checkoutMapper::toVm)
+                .collect(Collectors.toSet());
 
-        List<CheckoutItem> savedCheckoutItems = checkoutItemRepository.saveAll(checkoutItemList);
+        return checkoutVm.toBuilder().checkoutItemVms(checkoutItemVms).build();
+    }
 
-        Set<CheckoutItemVm> checkoutItemVms = savedCheckoutItems
-            .stream()
-            .map(checkoutMapper::toVm)
-            .collect(Collectors.toSet());
+    private Checkout setCheckoutItems(Checkout checkout, CheckoutPostVm checkoutPostVm) {
+        Set<Long> productIds = new HashSet<>();
+        List<CheckoutItem> checkoutItems = checkoutPostVm.checkoutItemPostVms()
+                .stream()
+                .map(checkoutItemPostVm -> {
+                    CheckoutItem item = checkoutMapper.toModel(checkoutItemPostVm);
+                    item.setCheckout(checkout);
+                    productIds.add(item.getProductId());
+                    return item;
+                })
+                .toList();
+
+        Map<Long, ProductCheckoutListVm> products = 
+                productService.getProductInfomation(productIds, 0, productIds.size());
+        associateProductWithCheckoutItem(checkout, products, checkoutItems);
+
+        checkout.setCheckoutItems(checkoutItems);
+        return checkout;
+    }
 
-        return checkoutVm.toBuilder().checkoutItemVms(checkoutItemVms).build();
+    private void associateProductWithCheckoutItem(
+            Checkout checkout,
+            Map<Long, ProductCheckoutListVm> products,
+            List<CheckoutItem> checkoutItems) {
+        checkoutItems.forEach(t -> {
+            ProductCheckoutListVm product = products.get(t.getProductId());
+            if (product == null) {
+                throw new NotFoundException("PRODUCT_NOT_FOUND", t.getProductId());
+            } else {
+                t.setProductName(product.getName());
+                t.setProductPrice(BigDecimal.valueOf(product.getPrice()));
+                checkout.addAmount(product.getPrice() * t.getQuantity());
```

Comment:
In my opinion, we should create a service responsible for calculating the total amount for `cart`, `checkout`, `order`, etc. This would help us manage the logic more easily and keep it consistent.

Reply:
I think this should have its own issue regarded calculate total amount base on tax, price, quantity, location, etc.